custom_components/tahoma/__init__.py

```diff
@@ -199,6 +205,36 @@ async def handle_execute_command(call):
         ),
     )
 
+    device_registry = await dr.async_get_registry(hass)
+
+    for gateway in gateways:
+        _LOGGER.debug(
+            "Gateway detected (%s - %s - %s)",
+            gateway.id,
+            gateway.type,
+            gateway.sub_type,
+        )
+
+        gateway_model = (
+            beautify_name(gateway.sub_type.name)
+            if isinstance(gateway.sub_type, Enum)
```

Comment:
It can be something else than an Enum?

Reply:
Unfortunately it can be something else... (https://github.com/iMicknl/python-tahoma-api/blob/78587f4f9547b2621a1123817efb2f2a66f21d9b/pyhoma/models.py#L394-L402).

However, maybe we should address it there. And throw a None + warning that we don't have the value in Enum.

Reply:
In Pyhoma the code looks good. But here it sounds weird :S

Reply:
I can also do this in `beautify_name()`, but I would prefer to just pass a string to that to keep is reusable.